src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -239,6 +243,45 @@ private static void createBranches(LfNetwork lfNetwork, LoadingContext loadingCo
                 legNumber++;
             }
         }
+
+        List<Set<LfBus>> connectedSets = getNonImpedantConnectedSets(lfNetwork);
+        connectedSets.stream()
+            .filter(set -> set.size() > 2) // Two LfBuses with one voltage control is never a problem
+            .forEach(set -> set.forEach(b -> removeConflictuousDiscreteVoltageControlled(b, set)));
+    }
+
+    private static List<Set<LfBus>> getNonImpedantConnectedSets(LfNetwork network) {
```

Comment:
Can we refactor this part to make a common with what we use to deal with non impedant line in the equation system ?